stlab/channel.hpp

```diff
@@ -434,6 +448,10 @@ struct shared_process : shared_process_receiver<yield_type<T, Arg>>,
         bool do_run;
         {
             std::unique_lock<std::mutex> lock(_process_mutex);
+            while (_process_queue_size != 0 && _process_queue_size <= _process_message_queue.size())
+            {
+                _process_queue_condition.wait(lock);
```

Comment:
No blocking allowed! This is a deadlock on a single threaded system and can cause a deadlock within the thread pool (at this point we now have one thread removed from the pool). The behavior of the queue limit is to control the cts (clear to send) signal which schedules upstream tasks. I do not believe this condition variable is necessary.